_posts/2025-04-07-introducing-quarkus-backstage.adoc

```diff
@@ -0,0 +1,335 @@
+be added to git, pushed to a git repository and then registered in Backstage.
+
+Pretty much any Backstage installation should be able to read the component from github.
+To do the registration, starting from the Home screen:
+
+* Click on the "Create" button
+* Click on "Register existing component"
+* Paste the link to the `catalog-info.yaml` file in the git repository
+* Click on "Analyze"
+* Click on "Register"  
+
+The registered component should look like:
+
+image::hello-backstage-component-view.png[scaledwidth=100%]
+
+This is not very impressive given that the file could be generated by Backstage itself.
+However, that would be the last thing that Backstage would do for you. 
+What would happen next time your application changes and changes need to be reflected in Backstage ?
+
+Let's expose an API for our application. We'll add the openapi extension and we'll configure it so
+that it generates the openapi schema in the same directory as the `catalog-info.yaml` file.
+
+[source,shell]
+----
+quarkus ext add smallrye-openapi
+echo "quarkus.smallrye-openapi.store-schema-directory=." >> src/main/resources/application.properties 
+----
+
+....
+Looking for the newly published extensions in registry.quarkus.io
+[SUCCESS] ✅  Extension io.quarkus:quarkus-smallrye-openapi has been installed
+....
+
+
+Let's rebuild the application and see how our `catalog-info.yaml` file looks like:
+
+[source,shell]
+----
+cat catalog-info.yaml
+----
+
+....
+ 
+apiVersion: backstage.io/v1alpha1
+kind: API
+spec:
+  type: openapi
+  lifecycle: production
+  owner: user:guest
+  definition:
+    $text: ./openapi.yaml
+metadata:
+  name: hello-backstage-api
+--- 
+apiVersion: backstage.io/v1alpha1
+kind: Component
+spec:
+  type: application
+  lifecycle: production
+  owner: user:guest
+  providesApis:
+  - hello-backstage-api
+metadata:
+  name: hello-backstage
+  labels:
+    app.kubernetes.io/name: hello-backstage
+    app.kubernetes.io/version: 1.0.0-SNAPSHOT
+    app.quarkus.io/version: 3.18.3
+  tags:
+  - java
+  - quarkus
+....
+
+The refreshed component now looks like:
+
+image::hello-backstage-component-with-api-view.png[scaledwidth=100%]
+
+
+==== As an extension ====
+An alternative to using the https://docs.quarkiverse.io/quarkus-backstage/dev/index.html[Quarkus Backstage] CLI plugin is to use the https://docs.quarkiverse.io/quarkus-backstage/dev/index.html[Quarkus Backstage] extension.
+Usually, an extension is needed when the application itself requires a library, a framework, or a code
+generator. In this case, the extension adds the Backstage entity generation to the application build process.
+
+To add the extension to the project:
+
+[source,shell]
+----
+quarkus ext add io.quarkiverse.backstage:quarkus-backstage:0.6.0
+----
+
+....
+[SUCCESS] ✅  Extension io.quarkiverse.backstage:quarkus-backstage:0.6.0 has been installed
+....
+
+
+At the time of writing the extension is not part of the Extension catalog, and thus it needs to be added using
```

Comment:
is it added by now?

Reply:
Yes, let's get it in. It's so easy to do ...